src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStringContainsOverIndexOfTests.cs

```diff
@@ -0,0 +1,707 @@
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.Preview,
+            };
+            await testCulture.RunAsync();
+
+            operatorKind = operatorKind == " == " ? " = " : operatorKind;
+            notString = notString == "!" ? " Not" : notString;
+            string vbInput = @"
+Public Class StringOf
+    Class TestClass
+        Public Sub AMethod(arg As String)
+            If [|arg.IndexOf(""This"")" + operatorKind + value + @"|] Then
+
+            End If
+        End Sub
+    End Class
+End Class
+";
+
+            var testOrdinal_vb = new VerifyVB.Test
+            {
+                TestState = { Sources = { vbInput } },
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp50,
+            };
+            await testOrdinal_vb.RunAsync();
+        }
+
+        [Theory]
+        [InlineData(" == ", " -1", "!")]
+        [InlineData(" >= ", " 0", "")]
+        public async Task TestFunctionParameterWithStringComparisonArgument(string operatorKind, string value, string notString)
+        {
+            string csInput = @"
+namespace TestNamespace 
+{ 
+    class TestClass 
+    { 
+        private void TestMethod(string str, System.StringComparison comparison) 
+        { 
+            if ([|str.IndexOf(""This"", comparison)" + operatorKind + value + @"|])
+            {
+            }
+        } 
+    } 
+}";
+            var testCulture = new VerifyCS.Test
+            {
+                TestState = { Sources = { csInput } },
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp50,
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.Preview,
+            };
+            await testCulture.RunAsync();
+
+            operatorKind = operatorKind == " == " ? " = " : operatorKind;
+            notString = notString == "!" ? " Not" : notString;
+            string vbInput = @"
+Public Class StringOf
+    Class TestClass
+        Public Sub AMethod(arg As String, comparison As System.StringComparison)
+            If [|arg.IndexOf(""This"", comparison)" + operatorKind + value + @"|] Then
+
+            End If
+        End Sub
+    End Class
+End Class
+";
+
+            var testOrdinal_vb = new VerifyVB.Test
+            {
+                TestState = { Sources = { vbInput } },
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp50,
+            };
+            await testOrdinal_vb.RunAsync();
+        }
+
+        [Theory]
+        [InlineData(" == ", " -1", "!")]
+        [InlineData(" >= ", " 0", "")]
+        public async Task TestMultipleDeclarations(string operatorKind, string value, string notString)
+        {
+            string csInput = @"
+namespace TestNamespace 
+{ 
+    class TestClass 
+    { 
+        private void TestMethod() 
+        { 
+            const string str = ""This is a string"";
+            int index = [|str.IndexOf('a', System.StringComparison.Ordinal)|], aa = 5;
```

Comment:
Also add a test where you have reverse order of declarators, i.e. `int aa = 5, index = [|str.IndexOf('a', System.StringComparison.Ordinal)|];`

Reply:
Done. Also modified a unit test to have `(-1 == str.IndexOf)`